### PR TITLE
Use a method per subdivision approach for [a-i] countries

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change

Update countries to align with a method per subdivision scheme [a-i]:
  - Andorra (migrated to holiday groups)
  - Bolivia (migrated to holiday groups)
  - Brazil (migrated to holiday groups)
  - Chile
  - France (migrated to alpha-2 subdivisions)
  - India
  - Italy (migrated to holiday groups)

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/italy.py [hunk 1]
```diff
@@ -144,270 +142,431 @@ class Italy(HolidayBase):
...
+
+    def _add_subdiv_fg_holidays(self):
+        self._add_holiday("Madonna dei Sette Veli", MAR, 22)
+
+    def _add_subdiv_fi_holidays(self):
+        self._add_holiday("San Giovanni Battista", JUN, 24)
+
+    def _add_subdiv_fm_holidays(self):
+        self._add_holiday("Maria Santissima Assunta", AUG, 15)
+        self._add_holiday("Maria Santissima Assunta", AUG, 16)
```
Comment (KJhellico) [inline holidays/countries/italy.py:327]:
```suggestion
        aug_15 = self._add_assumption_of_mary_day("Maria Santissima Assunta")
        self._add_holiday("Maria Santissima Assunta", aug_15 + td(days=+1))
```

File: holidays/countries/italy.py [hunk 2]
```diff
@@ -144,270 +142,431 @@ class Italy(HolidayBase):
...
+
+    def _add_subdiv_fr_holidays(self):
+        self._add_holiday("San Silverio", JUN, 20)
+
+    def _add_subdiv_ge_holidays(self):
+        self._add_holiday("San Giovanni Battista", JUN, 24)
```
Comment (KJhellico) [inline holidays/countries/italy.py:333]:
```suggestion
        self._add_saint_johns_day("San Giovanni Battista")
```

File: holidays/countries/italy.py [hunk 3]
```diff
@@ -144,270 +142,431 @@ class Italy(HolidayBase):
...
+        self._add_holiday("San Bassiano", JAN, 19)
+
+    def _add_subdiv_lt_holidays(self):
+        self._add_holiday("San Marco evangelista", APR, 25)
+
+    def _add_subdiv_lu_holidays(self):
+        self._add_holiday("San Paolino di Lucca", JUL, 12)
+
+    def _add_subdiv_mb_holidays(self):
+        self._add_holiday("San Giovanni Battista", JUN, 24)
```
Comment (KJhellico) [inline holidays/countries/italy.py:369]:
```suggestion
        self._add_saint_johns_day("San Giovanni Battista")
```

File: holidays/countries/italy.py [hunk 4]
```diff
@@ -144,270 +142,431 @@ class Italy(HolidayBase):
...
+
+    def _add_subdiv_pg_holidays(self):
+        self._add_holiday("Sant'Ercolano e San Lorenzo", JAN, 29)
+
+    def _add_subdiv_pi_holidays(self):
+        self._add_holiday("San Ranieri", JUN, 17)
+
+    def _add_subdiv_pn_holidays(self):
+        self._add_holiday("San Marco Evangelista", APR, 25)
+        self._add_holiday("Madonna delle Grazie", SEP, 8)
```
Comment (KJhellico) [inline holidays/countries/italy.py:424]:
```suggestion
        self._add_nativity_of_mary_day("Madonna delle Grazie")
```

File: holidays/countries/italy.py [hunk 5]
```diff
@@ -144,270 +142,431 @@ class Italy(HolidayBase):
...
+
+    def _add_subdiv_po_holidays(self):
+        self._add_holiday("Santo Stefano", DEC, 26)
```
Comment (KJhellico) [inline holidays/countries/italy.py:427]:
```suggestion
        self._add_christmas_day_two("Santo Stefano")
```

File: holidays/countries/italy.py [hunk 6]
```diff
@@ -144,270 +142,431 @@ class Italy(HolidayBase):
...
+        self._add_holiday("San Prospero Vescovo", NOV, 24)
+
+    def _add_subdiv_rg_holidays(self):
+        self._add_holiday("San Giorgio", APR, 23)
+
+    def _add_subdiv_ri_holidays(self):
+        self._add_holiday("Santa Barbara", DEC, 4)
+
+    def _add_subdiv_rm_holidays(self):
+        self._add_holiday("Santi Pietro e Paolo", JUN, 29)
```
Comment (KJhellico) [inline holidays/countries/italy.py:461]:
```suggestion
        self._add_saints_peter_and_paul_day("Santi Pietro e Paolo")
```

File: holidays/countries/italy.py [hunk 7]
```diff
@@ -144,270 +142,431 @@ class Italy(HolidayBase):
...
+        self._add_holiday("San Matteo Evangelista", SEP, 21)
+
+    def _add_subdiv_si_holidays(self):
+        self._add_holiday("Sant'Ansano", DEC, 1)
+
+    def _add_subdiv_so_holidays(self):
+        self._add_holiday("San Gervasio e San Protasio", JUN, 19)
+
+    def _add_subdiv_sp_holidays(self):
+        self._add_holiday("San Giuseppe", MAR, 19)
```
Comment (KJhellico) [inline holidays/countries/italy.py:479]:
```suggestion
        self._add_saint_josephs_day("San Giuseppe")
```

File: holidays/countries/italy.py [hunk 8]
```diff
@@ -144,270 +142,431 @@ class Italy(HolidayBase):
...
+        self._add_holiday("San Cataldo", MAY, 10)
+
+    def _add_subdiv_te_holidays(self):
+        self._add_holiday("San Berardo da Pagliara", DEC, 19)
+
+    def _add_subdiv_tn_holidays(self):
+        self._add_holiday("San Vigilio", JUN, 26)
+
+    def _add_subdiv_to_holidays(self):
+        self._add_holiday("San Giovanni Battista", JUN, 24)
```
Comment (KJhellico) [inline holidays/countries/italy.py:506]:
```suggestion
        self._add_saint_johns_day("San Giovanni Battista")
```

File: holidays/countries/italy.py [hunk 9]
```diff
@@ -144,270 +142,431 @@ class Italy(HolidayBase):
...
+        self._add_holiday(
+            "San Riccardo di Andria",
+            _get_nth_weekday_of_month(3, SUN, SEP, self._year),
+        )
+
+    def _add_subdiv_barletta_holidays(self):
+        self._add_holiday("San Ruggero", DEC, 30)
+
+    def _add_subdiv_cesena_holidays(self):
+        self._add_holiday("San Giovanni Battista", JUN, 24)
```
Comment (KJhellico) [inline holidays/countries/italy.py:557]:
```suggestion
        self._add_saint_johns_day("San Giovanni Battista")
```

**Review by arkid15r — COMMENTED** [review]
Thanks for the review!

**Review by KJhellico — COMMENTED** [review]
LGTM 👍

**Pull request merged**